docs/reference/search/knn-search.asciidoc

```diff
@@ -0,0 +1,129 @@
+[[knn-search]]
+=== kNN search API
+++++
+<titleabbrev>kNN search</titleabbrev>
+++++
+
+experimental::[]
+
+Performs a k-nearest neighbor search and returns the matching documents.
+
+[source,console]
+----
+PUT my-vector-index
+{
+  "mappings": {
+    "properties": {
+      "image_vector": {
+        "type": "dense_vector",
+        "dims": 3,
+        "index": true,
+        "similarity": "l2_norm"
+      }
+    }
+  }
+}
+
+GET /my-vector-index/_knn_search
+{
+  "knn": {
+    "field": "image_vector",
+    "query_vector": [0.3, 0.1, 1.2],
+    "k": 10,
+    "num_candidates": 100
+  },
+  "_source": ["name", "date"]
+}
+----
+// TEST[setup:my_index]
```

Comment:
Thanks for catching these issues! I ended up hiding the mapping set-up.

> Is there a reason we went with _source instead of fields? Not a huge deal, but I figured fields would be preferred.

I initially had `fields` in some API examples, and one of the readers got very confused. Approaching the API for the first time, they thought `fields` referred to the vector fields for kNN. Unfortunately `fields` is not a widely-used parameter yet, so users would be learning two different concepts at the same time.